src/sentry/SentryTransactionHub.ts

```diff
@@ -0,0 +1,27 @@
+import * as Sentry from '@sentry/react-native'
+import { Transaction } from '@sentry/types'
+import { SentrySpan, SentrySpans } from 'src/sentry/SentrySpans'
+
+let transactions = [] as Array<Transaction>
+
+export const SentryTransactionHub = {
+  startTransaction(name: SentrySpan) {
+    const transaction = Sentry.startTransaction({ ...SentrySpans[name], trimEnd: true })
+    transactions.push(transaction)
+  },
+  finishTransaction(name: SentrySpan) {
+    // get span operation - 'op'
+    const op = SentrySpans[name].op
+
+    // Find all the transactions with this op.
+    const selectedTransactions = transactions.filter(
```

Comment:
I don't understand this part.  Does it imply `startTransaction` is called with a `name` multiple times, but there isn't a always matching calls to `finishTransaction`?  Why not?

Reply:
`startTransaction` transaction could be called on a span that is never completed such as a wallet connect transaction that results in an error. This finishes and then removes all other spans of that same name from the SentryTransactionHub.